lib/util/Map.v3

```diff
@@ -110,6 +110,36 @@ class HashMap<K, V> extends PartialMap<K, V> {
 			}
 		}
 	}
+	// Number of key/value pairs in HashMap (expensive operation - cache the value).
+	def length() -> int {
+		if (table == null) {
+			// zero or one entry
+			if (cache != null) return 1;
+			return 0;
+		}
+		// two or more entries
+		var r = 0;
+		for (b in table) {
+			for (bucket = b; bucket != null; bucket = bucket.next) { r++; }
+		}
+    return r;
+	}
+	// Apply {func} to every (key, value) pair in this map with index.
+	def applyWithIndex(func: (K, V, int) -> void) {
```

Comment:
Similarly here, I think this is enough of a special case I'd rather leave it to user code for now.